modules/xmpp/VideoSIPGW.js

```diff
@@ -0,0 +1,184 @@
+
+        // check for global availability of the service
+        if (newStatus !== this.status
+            && (newStatus === Constants.STATUS_UNDEFINED
+                || newStatus === Constants.STATUS_AVAILABLE
+                || newStatus === Constants.STATUS_BUSY)) {
+            this.status = newStatus;
+            this.eventEmitter.emit(
+                XMPPEvents.VIDEO_SIP_GW_AVAILABILITY_CHANGED, this.status);
+        }
+    }
+
+    /**
+     * Handles presence nodes with name: jibri-sip-call-state.
+     *
+     * @param node the presence node to handle.
+     */
+    handleJibriSIPState(node) {
+        const attributes = node.attributes;
+
+        if (!attributes) {
+            return;
+        }
+
+        logger.log('Handle video sip gw state : ', attributes);
+
+        const newState = attributes.state;
+
+        if (newState === this.state) {
+            return;
+        }
+
+        switch (newState) {
+        case Constants.STATE_ON:
+        case Constants.STATE_OFF:
+        case Constants.STATE_PENDING:
+        case Constants.STATE_RETRYING:
+        case Constants.STATE_FAILED: {
+            const address = attributes.sipaddress;
+
+            if (!address) {
+                return;
+            }
+
+            // find the corresponding session and set its state
+            const session = this.sessions[address];
+
+            if (session) {
+                session.setState(newState);
+            } else {
+                logger.warn('Video SIP GW session not found:', address);
+            }
+        }
+        }
+    }
+
+
+        /**
+     * Creates new session and stores its reference.
+     *
+     * @param sipAddress - The sip address to use.
+     * @param displayName - The display name to use.
+     * @returns {JitsiVideoSIPGWSession}
+     */
+    createVideoSIPGWSession(sipAddress, displayName) {
+        const session = new JitsiVideoSIPGWSession(
+            sipAddress, displayName, this.chatRoom);
+
+        session.addStateListener(this.sessionStateChangeListener);
+        this.sessions[sipAddress] = session;
+
+        return session;
+    }
+
+    /**
+     * Returns whether SIP GW service is available.
+     *
+     * @returns {boolean} whether SIP GW service is available.
+     */
+    isVideoSIPGWAvailable() {
+        return this.status === Constants.STATUS_AVAILABLE;
+    }
+
+    /**
+     * Listener for session state changed. When a session goes to off or failed
+     * we delete its reference.
+     *
+     * @param address - The SIP address of the session.
+     * @param event - {name, oldState, newState}
+     */
+    sessionStateChanged(address, event) {
+        if (event.newState === Constants.STATE_OFF
+            || event.newState === Constants.STATE_FAILED) {
+            const session = this.sessions[address];
+
+            session.removeStateListener(this.sessionStateChangeListener);
+            delete this.sessions[address];
```

Comment:
It would be worth to mention somewhere in the JSDoc that session instance once enters this OFF or FAILED should not be used again.